basic-api.yml

```diff
@@ -0,0 +1,142 @@
+components:
+  schemas:
+    IssueRequest:
+      type: object
+      properties:
+        credential:
+          type: object
+          description: The credential to be issued minus the proofs
+          example: {
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://w3id.org/citizenship/v1"
+  ],
+  "id": "https://issuer.oidp.uscis.gov/credentials/83627465",
+  "type": [
+    "VerifiableCredential",
+    "PermanentResidentCard"
+  ],
+  "issuanceDate": "2019-12-03T12:19:52Z",
+  "expirationDate": "2029-12-03T12:19:52Z",
+  "credentialSubject": {
+    "id": "did:example:b34ca6cd37bbf23",
+    "type": [
+      "PermanentResident",
+      "Person"
+    ],
+    "givenName": "JOHN",
+    "familyName": "SMITH",
+    "gender": "Male",
+    "image": "data:image/png;base64,iVBORw0KGgo...kJggg==",
+    "residentSince": "2015-01-01",
+    "lprCategory": "C09",
+    "lprNumber": "999-999-999",
+    "commuterClassification": "C1",
+    "birthCountry": "Bahamas",
+    "birthDate": "1958-07-17"
+  }
+}
+        options:
+          type: object
+          description: Optional values that check or modify the input and modify the output of the request.
+          properties:
+            validation:
+              type: object
+              description: Optional validation schemas to use on the input document.
+              properties:
+                type:
+                  type: string
+                  description: The type of validation mechanism
+                  example: JsonSchema2018
+                credentialSchema:
+                  type: string
+                  description: A link to a credential schema to use for validation.
+                  example: https://example.com/schemas/prc.json
+            issuer:
+              type: string
+              description: Optional identifier for the issuer of the credential. If this is missing, it may be pre-configured or implied by other values in the request.
+              example: did:example:ec712ebc6f1c221ebfeb1f76e12
+            output:
+              type: object
+              description: Optional values that modify the output of the request.
+              properties:
+                representation:
+                  description: Optional value to specify the credential representation format. Well-known values are "jsonld" and "json". Other values MAY be supported. If this option is not specified a default of "jsonld" MUST be assumed.
+                  example: jsonld
+                proof:
+                  type: object
+                  description: Optional settings that modify the output format for the cryptographic proof.
+                  properties:
+                    format:
+                      description: Optional value to specify the credential representation format. Well-known values are "ldproof" and "jwt". Other values MAY be supported. If this option is not specified a default of "ldproof" MUST be assumed.
```

Comment:
See above, the API should not have a built-in preference for a specific credential or proof format.

Reply:
the more I think about this, the more it feels like a required field.

Reply:
Yeah, if we're not going to set a sane default, we should require the field (which makes me slightly sad).